pkg/kubecost/asset.go

```diff
@@ -3096,14 +3166,57 @@ func (asr *AssetSetRange) End() (time.Time, error) {
 	return end, nil
 }
 
+// Each iterates over all AssetSets in the AssetSetRange and returns the start and end over
+// the entire range
+func (asr *AssetSetRange) StartAndEnd() (time.Time, time.Time, error) {
```

Comment:
The `Minutes()` function was calling `Start()` and `End()` back to back, which was requiring nested loops each call. There could also exist contention between each call. It made sense to combine these into a single loop, and add proper locking.